src/main/java/com/epam/spring/boot/cargodeliverysystem/repository/imp/ReceiptStatusRepositoryImpl.java

```diff
@@ -0,0 +1,50 @@
+package com.epam.spring.boot.cargodeliverysystem.repository.imp;
+
+import com.epam.spring.boot.cargodeliverysystem.model.ReceiptStatus;
+import com.epam.spring.boot.cargodeliverysystem.repository.ReceiptStatusRepository;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Slf4j
+@Component
+public class ReceiptStatusRepositoryImpl implements ReceiptStatusRepository {
+
+    private final List<ReceiptStatus> receiptStatusList = new ArrayList<>();
+
+    @Override
+    public ReceiptStatus getReceiptStatus(String statusName) {
+        log.info("[Repository] getReceiptStatus by name {} ", statusName);
+        return receiptStatusList.stream()
+                .filter(s -> s.getStatusName().equals(statusName))
+                .findFirst()
+                .orElseThrow(() -> new RuntimeException("Receipt Status is not found!"));
```

Comment:
I would recommend create a specific custom exception, for example ReceiptStatusNotFoundException instead of abstract RuntimeException

Reply:
custom exceptions it's a task for one of the next HW's